src/routes/admin/profiles/editProfile.ts

```diff
@@ -5,178 +5,132 @@
-      } else {
-        log.debug('No secrets manager configured. Storing in DB.')
-        log.debug('Password will be visible in plain text.')
       }
-    }
-
-    if (!errorReason && results > 0) {
-      // TBD: Status should be 204
-      res.status(200).end(PROFILE_UPDATE_SUCCESS(amtConfig.ProfileName))
-    } else {
-      if (errorReason) res.status(500).end(`${errorReason}`)
-      else res.status(404).end(PROFILE_NOT_FOUND(amtConfig.ProfileName))
+      // SQL Query > Insert Data
+      const results = await profilesDb.updateProfile(amtConfig)
+      // profile inserted  into db successfully. insert the secret into vault
+      if (oldConfig.AMTPassword !== null || oldConfig.MEBxPassword !== null) {
+        if (req.secretsManager) {
+          log.debug('Delete in vault') // User might be flipping from false to true which we dont know. So try deleting either way.
+          await req.secretsManager.deleteSecretWithPath(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}profiles/${amtConfig.ProfileName}`)
+          log.debug('Password deleted from vault')
+        }
+      }
+      if (results) {
+      // store the password sent into Vault
+        if (req.secretsManager && (!amtConfig.GenerateRandomPassword || !amtConfig.GenerateRandomMEBxPassword)) {
+          const data = { data: {} }
+          if (!amtConfig.GenerateRandomPassword) {
+            data.data[`${amtConfig.ProfileName}_DEVICE_AMT_PASSWORD`] = amtPwdBefore
+            log.debug('AMT Password written to vault')
+          }
+          if (!amtConfig.GenerateRandomMEBxPassword) {
+            data.data[`${amtConfig.ProfileName}_DEVICE_MEBX_PASSWORD`] = mebxPwdBefore
+            log.debug('MEBX Password written to vault')
+          }
+          await req.secretsManager.writeSecretWithObject(`${EnvReader.GlobalEnvConfig.VaultConfig.SecretsPath}profiles/${amtConfig.ProfileName}`, data)
+        }
+        log.info(`Updated AMT profile: ${newConfig.profileName}`)
+        res.status(204).end()
+      }
     }
   } catch (error) {
-    if (res.status) return
-    console.log(error)
-    res.status(500).end(PROFILE_ERROR(amtConfig.ProfileName))
+    log.error(`Failed to update AMT profile: ${newConfig.profileName}`, error)
+    res.status(500).end(API_UNEXPECTED_EXCEPTION(`UPDATE ${newConfig.profileName}`))
   }
 }
 
-function readBody (req, res): AMTConfig {
-  const config: AMTConfig = {} as AMTConfig
-  const body = req.body
-
-  if (typeof body.payload.generateRandomPassword === 'string') {
-    if (body.payload.generateRandomPassword.toLowerCase() === 'true') {
-      body.payload.generateRandomPassword = true
-    } else if (body.payload.generateRandomPassword.toLowerCase() === 'false') {
-      body.payload.generateRandomPassword = false
-    }
+function getUpdatedData (newConfig: any, oldConfig: AMTConfiguration): AMTConfiguration {
+  const amtConfig: AMTConfiguration = {} as AMTConfiguration
```

Comment:
Just another way to write it -- optional:
``` js
 const amtConfig: AMTConfiguration = {
ProfileName: newConfig.profileName
} as AMTConfiguration
```

Reply:
done